### PR TITLE
DFAST-131: Configure black package for automating formatting

### PR DESCRIPTION
**Pull request by ralph-peelen-deltares**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -28,8 +28,37 @@ imageio = "^2.33.1"
...
+isort = "^5.11.4"
 teamcity-messages = "^1.32"
 mock = "^5.1.0"
+flake8 = "^6.0.0"
+pylint = "^2.15.10"
+
+[tool.black]
+line-length = 88
+target-version = ['py39']
+exclude = '''
```
Comment (MRVermeulenDeltares) [inline pyproject.toml:40]: can you explain this for me face to face?

File: pyproject.toml [hunk 2]
```diff
@@ -27,10 +27,37 @@ Nuitka = "^1.3.8"
...
 pytest-cov = "^4.1.0"
-black = "^22.12.0"
+black = "^24.2.0"
+isort = "^5.13.2"
 teamcity-messages = "^1.32"
 mock = "^5.1.0"
 
+[tool.black]
+line-length = 88
+target-version = ['py312']
```
Comment (MRVermeulenDeltares) [inline pyproject.toml:37]: why py312 if the yml makes use of 3.9?

File: .github/workflows/formatting.yml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+      - name: Install Dependencies
+        run: poetry install
+
+      - name: Autoformat code if the check fails
+        run: |
+          poetry run isort .
+          poetry run black .
+          git config --global user.name '${{ github.actor }}'
+          git config --global user.email '${{ github.actor }}@users.noreply.github.com'
+          git remote set-url origin https://x-access-token:${{ secrets.GITHUB_TOKEN }}@github.com/$GITHUB_REPOSITORY
```
Comment (MRVermeulenDeltares) [inline .github/workflows/formatting.yml:37]: Did you generate these tokens yourself or are they generated by a general deltares account?

Comment (priscavdsluis) [reply .github/workflows/formatting.yml:37]: No, I did not generate these myself, they are provided within the Deltares organization.

File: pyproject.toml (hunk 2 above)
Comment (priscavdsluis) [reply pyproject.toml:37]: I will change it to 3.9!

File: pyproject.toml (hunk 1 above)
Comment (priscavdsluis) [reply pyproject.toml:38]: This list contains the directories that should be skipped by black, for example you do not want to format external packages' code in your .venv directory, but only your own source code.

File: .github/workflows/formatting.yml (hunk 3 above)
Comment (MRVermeulenDeltares) [reply .github/workflows/formatting.yml:37]: nice!

File: pyproject.toml (hunk 1 above)
Comment (MRVermeulenDeltares) [reply pyproject.toml:38]: thx!

**Review by MRVermeulenDeltares — APPROVED** [review]
Had a look at the formatting.yml & pyproject.toml.
Looks good, as discussed, lets wait with merging untill bigger issues are merged. (123 & 125)

**Pull request merged**